lib/core/proxy.c

```diff
@@ -136,7 +136,7 @@ static h2o_iovec_t build_content_length(h2o_mem_pool_t *pool, size_t cl)
 {
     h2o_iovec_t cl_buf;
     cl_buf.base = h2o_mem_alloc_pool(pool, char, sizeof(H2O_UINT64_LONGEST_STR) - 1);
-    cl_buf.len = sprintf(cl_buf.base, "%zu", cl);
+    cl_buf.len = sprintf(cl_buf.base, "%" PRIu64, cl);
```

Comment:
Thank you for the PR, but this code is actually incorrect, the outcome is undefined when sizeof(cl) != sizeof(uint64_t) due to how `cl` is being pushed onto the stack.

I now think that we should provide a proper fix, see #2205.

Reply:
I see your point, and thank you so much for looking further into this. I like #2205.